db/dcrpg/sync.go

```diff
@@ -98,19 +97,19 @@ const (
 // which the caller should wait to receive the result. As such, this method
 // should be called as a goroutine or it will hang on send if the channel is
 // unbuffered.
-func (db *ChainDB) SyncChainDBAsync(ctx context.Context, res chan dbtypes.SyncResult,
-	client rpcutils.MasterBlockGetter, updateAllAddresses, updateAllVotes, newIndexes bool,
+func (pgb *ChainDB) SyncChainDBAsync(ctx context.Context, res chan dbtypes.SyncResult,
+	client rpcutils.MasterBlockGetter, updateAllAddresses, newIndexes bool,
 	updateExplorer chan *chainhash.Hash, barLoad chan *dbtypes.ProgressBarLoad) {
-	if db == nil {
+	if pgb == nil {
 		res <- dbtypes.SyncResult{
 			Height: -1,
 			Error:  fmt.Errorf("ChainDB (psql) disabled"),
 		}
 		return
 	}
 
-	height, err := db.SyncChainDB(ctx, client, updateAllAddresses,
-		updateAllVotes, newIndexes, updateExplorer, barLoad)
+	height, err := pgb.SyncChainDB(ctx, client, updateAllAddresses, newIndexes,
+		updateExplorer, barLoad)
 	if err != nil {
 		log.Debugf("SyncChainDB quit at height %d, err: %v", height, err)
```

Comment:
I think this will be more effective if it was
```Golang
   log.Errorf("SyncChainDB quit at height %d, err: %v", height, err)
```

Reply:
Agreed.

Regarding the error `InsertVoutsStmt: pq: error in argument for $9: only 1-dimension arrays supported`, that's surprising since I recall resolving the exact error while preparing the PR.  What cockroach version are you using?  I'll have to check my local repo clone for unpublished commits.

Reply:
`2019-05-04 00:42:49.089 [INF] PSQL: CockroachDB CCL v19.1.0 (x86_64-unknown-linux-gnu, built 2019/04/29 18:36:40, go1.11.6)`

Reply:
The version I've been using:
`CockroachDB CCL v2.1.6 (x86_64-pc-linux-gnu, built 2019/04/21 02:11:11, go1.12.4)`

How bizarre that they went from 2.1.6 to 19.0 (just released on Apr. 30).

Anyway, I just downloaded crdb v19.0 and was able to replicate the issue.

Reply:
Looks like a bug in either github.com/lib/pq or CockroachDB v19.0.  It seems impossible to insert an empty `TEXT[]` with the lib/pq driver in crdb v19.0 although it worked with crdb v2.1.6.

Reply:
How about adding a recommendation that our cockroach db implementation only works with v2.1.6 to the wiki? You could also add a link on how you downloaded it.

Reply:
Turns out it was a regression in CockroachDB v19: https://github.com/cockroachdb/cockroach/issues/37365#issuecomment-490238226  A fix is reportedly planned.

Reply:
The fix was merged into cockroach master and is being back ported to 19.1.x: https://github.com/cockroachdb/cockroach/pull/37398
For now I've updated [the wiki](https://github.com/decred/dcrdata/wiki/CockroachDB).